crates/core_simd/src/first.rs

```diff
@@ -24,51 +24,166 @@ macro_rules! impl_vector {
 
             /// Converts a SIMD vector to an array.
             pub const fn to_array(self) -> [$type; LANES] {
-                // workaround for rust-lang/rust#80108
-                // TODO fix this
-                #[cfg(target_arch = "wasm32")]
-                {
-                    let mut arr = [self.0[0]; LANES];
-                    let mut i = 0;
-                    while i < LANES {
-                        arr[i] = self.0[i];
-                        i += 1;
-                    }
-                    arr
-                }
+                self.0
+            }
+        }
+
+        impl<const LANES: usize> $name<LANES>
+        where
+            Self: crate::Vector,
+            crate::MaskSize<LANES>: crate::Mask,
+            crate::SimdIsize<LANES>: crate::Vector,
+            crate::SimdUsize<LANES>: crate::Vector,
+        {
```

Comment:
I think we may be capable of getting rid of this file, which is an undesirable aberration in our implementation, soon, so I would prefer these remain factored out of this file, regardless of how this is achieved.

Reply:
I originally tried combining it into the root of the vector module, but it wasn't obvious that there was an order between the submodules and macros.  I moved the file into `vector/vector_impl.rs` and was able to move the `Vector` trait implementation into it, too.